static/js/containers/pages/CheckoutPage_test.js

```diff
@@ -102,6 +104,132 @@ describe("CheckoutPage", () => {
     assert.equal(inner.find("img").prop("alt"), basketItem.description)
     assert.equal(inner.find(".item-row .title").text(), basketItem.description)
   })
+  ;[true, false].forEach(hasError => {
+    it(`updates the basket with a product id from the query parameter${
+      hasError ? ", but an error is returned" : ""
+    }`, async () => {
+      const productId = 4567
+      if (hasError) {
+        helper.handleRequestStub.withArgs("/api/basket/", "PATCH").returns({
+          status: 400,
+          body:   {
+            errors: "error"
+          }
+        })
+      }
+      const { inner } = await renderPage(
+        {},
+        {
+          location: {
+            search: `product=${productId}`
+          }
+        }
+      )
+
+      sinon.assert.calledWith(
+        helper.handleRequestStub,
+        "/api/basket/",
+        "PATCH",
+        {
+          body:        { items: [{ id: productId }] },
+          credentials: undefined,
+          headers:     {
+            "X-CSRFTOKEN": null
+          }
+        }
+      )
+      assert.equal(inner.state().errors, hasError ? "error" : null)
+    })
+  })
+
+  //
+  ;[[true, false], [true, true], [false, false], [false, true]].forEach(
+    ([hasValidProductId, hasValidCoupon]) => {
+      it(`updates the basket with a ${
+        hasValidProductId ? "" : "in"
+      }valid product id and a ${
+        hasValidCoupon ? "" : "in"
+      }valid coupon code from the query parameter`, async () => {
+        const couponCode = "codeforcoupon"
+        const productId = 12345
+        const couponPayload = {
+          body:        { coupons: [{ code: couponCode }] },
+          credentials: undefined,
+          headers:     {
+            "X-CSRFTOKEN": null
+          }
+        }
+        const productPayload = {
+          body:        { items: [{ id: productId }] },
+          credentials: undefined,
+          headers:     {
+            "X-CSRFTOKEN": null
+          }
+        }
+
+        if (!hasValidProductId) {
+          helper.handleRequestStub
+            .withArgs("/api/basket/", "PATCH", productPayload)
+            .returns({
+              status: 400,
+              body:   {
+                errors: "product error"
+              }
+            })
+        }
+        if (!hasValidCoupon) {
+          helper.handleRequestStub
+            .withArgs("/api/basket/", "PATCH", couponPayload)
+            .returns({
+              status: 400,
+              body:   {
+                errors: "coupon error"
+              }
+            })
+        }
+        const { inner } = await renderPage(
+          {},
+          {
+            location: {
+              search: `product=${productId}&code=${couponCode}`
+            }
+          }
+        )
+        // wait for componentDidMount to resolve
+        await wait(15)
```

Comment:
Looks like there's no nice way to avoid this, but I'm wondering about option 2 in this GH post: https://github.com/airbnb/enzyme/issues/1587#issuecomment-416610674. If that seems to work, I think I'd prefer it over the `wait`